src/runtime/builtin_modules/builtins.cpp

```diff
@@ -1242,6 +1241,7 @@ Box* zip(BoxedTuple* containers) {
             el->elts[i] = *iterators[i];
             ++(iterators[i]);
         }
+        AUTO_DECREF(el);
```

Comment:
This is minor, but this should go right after the `BoxedTuple::create` call.  If any of the iterators throw an exception (their `++` can do that) then we would want to free the tuple.